src/hooks/api/useNoteEsc.js

```diff
@@ -0,0 +1,53 @@
+import { useQuery } from 'react-query';
+import { useIntl } from 'react-intl';
+import { useMemo } from 'react';
+
+import { useNamespace } from '@folio/stripes/core';
+
+import { getMappedAndSortedNotes } from '../../utils/helpers';
+import { usePublishCoordinator } from '../usePublishCoordinator';
+
+const DEFAULT_DATA = {};
+
+export const useNotesEsc = (namespaceKey, tenants, type, categoryId, options = {}, url, noteKey, optionType, parameterKey) => {
+  const [namespace] = useNamespace({ key: namespaceKey });
+  const { initPublicationRequest } = usePublishCoordinator(namespace);
+  const { formatMessage } = useIntl();
+
+  const { data = DEFAULT_DATA, isFetching } = useQuery({
+    queryKey: [namespace, tenants, url, type],
+    queryFn: async () => {
+      const { publicationResults } = await initPublicationRequest({
+        url,
+        method: 'GET',
+        tenants,
+      });
+      return publicationResults;
+    },
+    keepPreviousData: true,
+    ...options
+  });
+
+  const notesEsc = useMemo(() => {
+    if (!data?.length || isFetching) return [];
+    const notes = data.flatMap(tenantData => {
```

Comment:
I suggest to move this logic in separate helper function

Reply:
it's already in separte function, that using for each notes from esc